src/Marten/Events/EventStoreExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Linq;
+using Marten.Linq;
+
+namespace Marten.Events
+{
+    public static class EventStoreExtensions
+    {
+        public static T AggregateTo<T>(this IMartenQueryable<IEvent> queryable) where T : class
+        {
+            var session = ((MartenLinqQueryable<IEvent>)queryable).MartenSession;
```

Comment:
Hate to do the downcast, but it's going to be fine here. The Baseline `.As<T>()` extension method makes that a little cleaner code.

Reply:
Fixed.